### PR TITLE
Background worker

### PR DESCRIPTION
**Pull request by rohitdatta**
**Adds a background worker running off Celery.**

**Changes proposed in this pull request:**

* Use Flask CLI to launch processes (replaces Flask Script: `python manage.py runserver`)
* Add Celery background tasks
* Use Celery background tasks for failed email sending

**Have you made sure to add:**
- [ ] Tests
- [ ] Documentation

**Deploy notes**

- Changes the development server launch command
- Adds in the need for changing production launch script

**Any Additional Information**

**_This is currently a work in progress and not finished. While there isn't a lot of code, this is a big change and should be reviewed carefully. Additionally, unit tests haven't been written yet._**

### CODE REVIEW

**Comment by fiatjaf** [comment]
Can you share `pip freeze`? Flask, click, celery and maybe other packages must be downloaded/upgraded and it's better to have the exact versions.

**Comment by rohitdatta** [comment]
Yeah I think it's failing CI because of a package I installed
```
alembic==0.8.10
amqp==2.2.2
billiard==3.5.0.3
celery==4.1.0
click==6.7
fakeredis==0.8.2
Flask==0.12
Flask-CDN==1.5.3
Flask-Cors==3.0.2
Flask-Limiter==0.9.3
Flask-Login==0.3.0
Flask-Migrate==2.0.3
Flask-Redis==0.0.6
Flask-Script==2.0.5
Flask-SQLAlchemy==2.1
Flask-Testing==0.6.1
funcsigs==1.0.2
future==0.16.0
gunicorn==19.6.0
hashids==1.0.2
httpretty==0.8.14
itsdangerous==0.24
Jinja2==2.8.dev0
kombu==4.1.0
limits==1.2.1
Mako==1.0.6
MarkupSafe==0.23
mock==2.0.0
ordereddict==1.1
Paste==2.0.3
pbr==1.10.0
psycopg2==2.6.2
python-dotenv==0.3.0
python-editor==1.0
pytz==2017.2
redis==2.10.5
requests==2.1.0
six==1.10.0
SQLAlchemy==1.0.12
stripe==1.55.0
structlog==16.1.0
unicodecsv==0.14.1
vine==1.1.4
Werkzeug==0.11.9
```

**Comment by fiatjaf** [comment]
Are we going to commit this full list (that includes subdependencies)?
I don't get it, Flask is at the same version I have here and still I don't have the `flask` command.

**Comment by fiatjaf** [comment]
Ok, I don't know how can I test if this celery job is running fine. I'll try more later.

**Comment by rohitdatta** [comment]
So the Flask CLI needs you to tell it about the app, which README should now have. I actually realized this branched off of #162 so perhaps we should move the discussion to there.

**Comment by colevscode** [comment]
This PR looks good to me. Can we merge?

File: formspree/users/views.py [hunk 1]
```diff
@@ -314,13 +323,8 @@ def stripe_webhook():
...
             customer_id = event['data']['object']['customer']
             customer = stripe.Customer.retrieve(customer_id)
-            g.log.info('User payment failed', account=customer.email)
-            send_email(to=customer.email,
-                       subject='[ACTION REQUIRED] Failed Payment for {} {}'.format(settings.SERVICE_NAME,
-                                                                           settings.UPGRADED_PLAN_NAME),
-                       text=render_template('email/payment-failed.txt'),
-                       html=render_template('email/payment-failed.html'),
-                       sender=settings.DEFAULT_SENDER)
+            send_failed_email.delay(4)
```
Comment (fiatjaf) [inline formspree/users/views.py]: Shouldn't it be `send_failed_email.delay(customer)` here, as per http://docs.celeryproject.org/en/latest/userguide/calling.html#basics?

Comment (rohitdatta) [reply formspree/users/views.py]: Yes, you're right, this was hard coding for testing I didn't remove

**Comment by rohitdatta** [comment]
Apologies for the mess, I tried to rebase off of master because this was so out of date, but I think that may have caused it to become worse 😬

**Comment by rohitdatta** [comment]
Please see #185 for a simpler diff

**Comment by fiatjaf** [comment]
Hey, I was in the middle of a rebase here! 

Ok, thanks.

**Pull request closed without merging**